backend/src/middlewares/update-user/update-user.middleware.ts

```diff
@@ -0,0 +1,73 @@
+import { ErrorMessage, updateUserSchema } from '@autoline/shared';
+import { ExceptionMessage } from '@common/enums/exception/exception';
+import { TypedRequestBody } from '@common/types/controller/controller';
+import { UpdateUser } from '@controllers/update-user/update-user.controller';
+import { prisma } from '@data/prisma-client';
+import { validatePassword } from '@services/password-validation/password-validation.service';
+import { NextFunction, Response } from 'express';
+import httpStatus from 'http-status-codes';
+
+const updateUserMiddleware = async (
+  req: TypedRequestBody<UpdateUser>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    await updateUserSchema.validate(req.body);
+    await emailExist(req);
+    await passwordValidate(req);
+
+    next();
+  } catch (err) {
+    const { message } = err as ErrorMessage;
+    console.error(err);
+    res.status(httpStatus.FORBIDDEN).json({ error: message });
+    next(err);
+  }
+};
```

Comment:
I wouldn't do the validation of the input values for the update in the middleware. It's better to be done in the service because the validation here can be a part of the business logic. Especially the prohibition to pass the already existing password

Reply:
So we can't pass the current password in the reqest?
Then how do I know that the user has entered the current password correctly?

Reply:
No, it's okay to pass it. But I just wouldn't check in the middleware